### PR TITLE
Default Sync for group methods

### LINKED ISSUE #263: Default groups list methods to sync, optional parameter for skipping sync
### Is your feature request related to a problem?

_No response_

### Describe the solution to the problem

_No response_

### Describe the uses cases for the feature

_No response_

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by cameronvoell**
Closes #263 

All tests are still passing even though all `group.sync()` and `conversation.syncGroups()` calls have been removed from tests.

### CODE REVIEW

File: src/lib/Group.ts
Comment (alexrisch) [inline src/lib/Group.ts:82]: Thoughts on using an object instead?
Could get confusing especially when adding pagination and sorting later on

**Review by alexrisch — APPROVED** [review]
Approving so I'm not blocking, but added a question

File: src/lib/Group.ts
Comment (cameronvoell) [reply src/lib/Group.ts:82]: Yea I agree (if im understanding correctly), once we have pagination and sorting, passing in an object would be more straightforward. My inclination is to wait until one of those is ready before adding here, so I'll merge for now with the idea we'll update when those are ready in upcoming PR 🙏

**Pull request merged**
